### PR TITLE
Docker setup

### PR DESCRIPTION
**Pull request by archers1234**
setup docker for python3, opengl, c++, cmake, and make.

Issue: https://github.com/projectpolygon/hypercube/issues/6

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+docs/
+
+.git/
+.gitignore
+.dockerignore
+
+LICENSE
+Dockerfile
+
```
Comment (ansh-projects) [inline .dockerignore:13]: might want to ignore build files as well, stuff in gitignore essentially. __pycache__, etc

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    apt-get install -y g++ && \
+    apt-get install -y make && \
+    apt-get install -y cmake && \
+    apt-get install -y libglu1-mesa-dev freeglut3-dev mesa-common-dev && \
+    apt-get install -y git && \
+    apt-get install -y libxrandr-dev && \
+    apt-get install -y libxinerama-dev && \
+    apt-get install -y libxcursor-dev && \
+    apt-get install -y libxi-dev &&  \
+    apt-get install -y mesa-utils && \
```
Comment (ansh-projects) [inline Dockerfile:16]: you can just do apt install -y apt-utils \ python3 \ g++ ....
to just do one apt install. It willl fail the whole thing if any fail

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    mkdir hypercube
```
Comment (ansh-projects) [inline Dockerfile:17]: WORKDIR hypercube creates it, you dont need mkdir

then you can do ``` add . .``` to add into the current forlder (hypercube)

File: .dockerignore (hunk 1 above)
Comment (archers1234) [reply .dockerignore:19]: updated to include the sections in the .gitignore

File: Dockerfile (hunk 2 above)
Comment (archers1234) [reply Dockerfile]: fixed

File: Dockerfile (hunk 3 above)
Comment (archers1234) [reply Dockerfile]: fixed

**Pull request merged**